libs/sdk-ui-kit/src/Header/HeaderAccount.tsx

```diff
@@ -1,73 +1,61 @@
-// (C) 2007-2024 GoodData Corporation
-import React, { PureComponent, ReactNode } from "react";
+// (C) 2007-2025 GoodData Corporation
+import React, { useState, useRef, ReactNode, useCallback } from "react";
 import cx from "classnames";
-import { injectIntl, FormattedMessage, WrappedComponentProps } from "react-intl";
+import { useIntl, FormattedMessage } from "react-intl";
 
 import { Overlay } from "../Overlay/index.js";
 import { Button } from "../Button/index.js";
 
-import { IHeaderAccountState, IHeaderMenuItem, IHeaderAccountProps } from "./typings.js";
+import { IHeaderMenuItem, IHeaderAccountProps } from "./typings.js";
+import { UiFocusTrap } from "../@ui/UiFocusTrap/UiFocusTrap.js";
+import { useId } from "../utils/useId.js";
 
-class WrappedHeaderAccount extends PureComponent<
-    IHeaderAccountProps & WrappedComponentProps,
-    IHeaderAccountState
-> {
-    static defaultProps: Pick<IHeaderAccountProps, "className" | "items" | "userName"> = {
-        className: "",
-        items: [],
-        userName: "",
-    };
-
-    constructor(props: IHeaderAccountProps & WrappedComponentProps) {
-        super(props);
-
-        this.state = {
-            isOpen: false,
-        };
-    }
+export const HeaderAccount: React.FC<IHeaderAccountProps> = ({
+    className = "",
+    items = [],
+    userName = "",
+    onMenuItemClick,
+}) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    const intl = useIntl();
+    const id = useId();
+    const dropdownId = `account-dropdown-${id}`;
 
-    getClassNames(): string {
+    const getClassNames = (): string => {
```

Comment:
Does this need to be wrapped in function? Imho `const classNames = cx(...);` should work the same?

Reply:
AI relict